mathesar/pagination.py

```diff
@@ -0,0 +1,29 @@
+from collections import OrderedDict
+
+from rest_framework.pagination import LimitOffsetPagination
+from rest_framework.response import Response
+
+
+class DefaultLimitOffsetPagination(LimitOffsetPagination):
+    default_limit = 50
+    max_limit = 500
+
+    def get_paginated_response(self, data):
+        return Response(OrderedDict([
+            ('count', self.count),
+            ('results', data)
+        ]))
+
+
+class TableLimitOffsetPagination(DefaultLimitOffsetPagination):
+
+    def paginate_queryset(self, queryset, request, table_id):
+        self.limit = self.get_limit(request)
+        if self.limit is None:
+            self.limit = self.default_limit
+        self.offset = self.get_offset(request)
+        # TODO: Cache count value somewhere, since calculating it is expensive.
```

Comment:
It's somewhat infeasible to have this stat for large tables (at least an accurate version of it).  What are we using it for?  Can it be avoided?  Would an approximate value suffice?  Note that the `select` being used in `get_records`  will just return an empty record set (no error) if `limit * offset > count` .  If `count` is needed to satisfy the `LimitOffsetPagination` class, I suggest we spoof it with a very large value, and simply notice/handle an empty result set ourselves.